src/api/offline/ledgerhash.js

```diff
@@ -28,25 +28,47 @@ function hashLedgerHeader(ledgerHeader) {
   return common.core.Ledger.calculateLedgerHash(header);
 }
 
-function computeLedgerHash(ledgerHeader: Object, transactions: Array<Object>
-): string {
-  if (transactions) {
-    const txs = _.map(transactions, tx => {
-      const mergeTx = _.assign({}, _.omit(tx, 'tx'), tx.tx || {});
-      const renameMeta = _.assign({}, _.omit(mergeTx, 'meta'),
-        tx.meta ? {metaData: tx.meta} : {});
-      return renameMeta;
-    });
-    const ledger = common.core.Ledger.from_json({transactions: txs});
-    const transactionHash = ledger.calc_tx_hash().to_hex();
-    if (ledgerHeader.transaction_hash !== undefined
-        && ledgerHeader.transaction_hash !== transactionHash) {
-      throw new common.errors.ValidationError('transaction_hash in header'
-        + ' does not match computed hash of transactions');
-    }
-    return hashLedgerHeader(_.assign({}, ledgerHeader, {transactionHash}));
+function computeTransactionHash(ledger) {
+  if (ledger.rawTransactions === undefined) {
+    return ledger.transactionHash;
   }
-  return hashLedgerHeader(ledgerHeader);
+  const transactions = JSON.parse(ledger.rawTransactions);
+  const txs = _.map(transactions, tx => {
+    const mergeTx = _.assign({}, _.omit(tx, 'tx'), tx.tx || {});
+    const renameMeta = _.assign({}, _.omit(mergeTx, 'meta'),
+      tx.meta ? {metaData: tx.meta} : {});
+    return renameMeta;
+  });
+  const ledgerObject = common.core.Ledger.from_json({transactions: txs});
+  const transactionHash = ledgerObject.calc_tx_hash().to_hex();
+  if (ledger.transactionHash !== undefined
+      && ledger.transactionHash !== transactionHash) {
+    throw new common.errors.ValidationError('transactionHash in header'
+      + ' does not match computed hash of transactions');
+  }
+  return transactionHash;
+}
+
+function computeStateHash(ledger) {
+  if (ledger.rawState === undefined) {
+    return ledger.stateHash;
+  }
+  const state = JSON.parse(ledger.rawState);
+  const ledgerObject = common.core.Ledger.from_json({accountState: state});
+  const stateHash = ledgerObject.calc_account_hash().to_hex();
+  if (ledger.stateHash !== undefined && ledger.stateHash !== stateHash) {
+    throw new common.errors.ValidationError('stateHash in header'
+      + ' does not match computed hash of state');
+  }
+  return stateHash;
+}
+
+function computeLedgerHash(ledger: Object): string {
```

Comment:
Do you have any specific ideas? We already have plans to create a Wallet module, which will likely be exported parallel to RippleAPI and would have generateAddress and isValidAddress. We also have plans to make more documentation, but that is lower priority until it's more stable. I don't have an idea for where to put computeLedgerHash yet.